servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp

```diff
@@ -172,6 +172,7 @@ RID RenderForwardMobile::RenderBufferDataForwardMobile::get_color_fbs(Framebuffe
 
 	uint32_t view_count = render_buffers->get_view_count();
 
+	bool vrs_uses_fdm = RD::get_singleton()->has_feature(RD::SUPPORTS_FRAGMENT_DENSITY_MAP);
```

Comment:
So just to hark back on my earlier remark up above. If we go for an API that always specifies a VRS attachment on the pass, and we just apply it to all subpasses, instead of adding an attachment to each subpass here, we can simplify this logic and not have to make choices here which may also end up being different on Metal and/or DirectX.

If we do keep the current approach of adding attachments to subpasses here, I think the constant passed here should be more generic, like `RD::SUPPORT_VRS_PER_SUBPASS` that switches whether we can specify a density map per subpass instead of supplying one for the whole pass.